functions/Get-DbaDbTLogSpace.ps1

```diff
@@ -0,0 +1,118 @@
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database, LogFile
+        Author: Jess Pomfret, JessPomfret.com
+
+        Website: https://dbatools.io
+        Copyright: (c) 2019 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaDbTLogSpace
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1
+
+        Returns the transaction log usage information for all databases on Server1
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -Database Database1, Database2
+
+        Returns the transaction log usage information for both Database1 and Database 2 on Server1
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -ExcludeDatabase Database3
+
+        Returns the transaction log usage information for all databases on Server1, except Database3
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbTLogSpace -SqlInstance Server1 -ExcludeSystemDatabase
+
+        Returns the transaction log usage information for all databases on Server1, except the system databases
+
+    .EXAMPLE
+        PS C:\> Get-DbaRegisteredServer -SqlInstance cmsServer | Get-DbaDbTLogSpace -Database Database1
+
+        Returns the transaction log usage information for Database1 for a group of servers from SQL Server Central Management Server (CMS).
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(ValueFromPipeline, Mandatory)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [switch]$ExcludeSystemDatabase,
+        [switch]$EnableException
+    )
+
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failed to process Instance $Instance" -ErrorRecord $_ -Target $instance -Continue
+            }
+            $dbs = $server.Databases | Where-Object IsAccessible
+
+            if ($Database) {
+                $dbs = $dbs | Where-Object Name -in $Database
+            }
+            if ($ExcludeDatabase) {
+                $dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+            }
+
+            if ($ExcludeSystemDatabase) {
+                $dbs = $dbs | Where-Object IsSystemObject -eq $false
+            }
+
+            try {
+                $logspace = Invoke-DbaQuery -SqlInstance $server -Query "dbcc sqlperf(logspace)" | Where-Object { $dbs.name -contains $_.'Database Name' }
```

Comment:
I'm on the fence on this one...will let @potatoqualitee chime in.

I'm for:
```suggestion
                $logspace = $server.Query("dbcc sqlperf(logspace)")
```

Reply:
I would also prefer we validate version and for 2012+ use the new DMV. [sys.dm_db_log_space_usage](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-log-space-usage-transact-sql).

This requires executing within the context of each database but also includes an additional property for determining if you need to take a log backup (`log_space_in_bytes_since_last_backup`):
![image](https://user-images.githubusercontent.com/11204251/62562203-5385a600-b846-11e9-9975-5cc762bec844.png)

Reply:
The image is against Azure SQL database as well.

Reply:
I wondered about checking for versions and using the newer dmv, chose to use dbcc instead to make it easy. I didn't think about Azure though - I'll add in this logic.

What's the benefit of using $server.Query over Invoke-DbaQuery?

Reply:
You have additional overhead using Invoke-DbaQuery because it does more than just `$server.Query()` will....it calls the async command which is overkill for this purpose.

Reply:
I can see responses coming about wanting the log space info since last log backup with the name of this command. If we do not want to add it now that is fine, it can just be added later on but I think it is needed. As well, the new DMV will actually return information for Azure SQL Database, where the DBCC command does not.

Reply:
Oh I'll add the logic for using the DMV today, and that makes sense to change to Server.Query. Was just curious for the reasoning, thanks!